src/Datadog.Trace.ClrProfiler.Managed.Loader/Startup.cs

```diff
@@ -1,52 +1,186 @@
 using System;
+using System.IO;
 using System.Reflection;
 
 namespace Datadog.Trace.ClrProfiler.Managed.Loader
 {
     /// <summary>
-    /// A class that attempts to load the Datadog.Trace.ClrProfiler.Managed .NET assembly.
+    /// This is the only publi class in this assembly.
+    /// The entire assembly is compiled into the native profiler DLL as a resource.
+    /// This happens both, for the profiler part of the Tracer, and for the actual Profiler.
+    /// The native code then uses this class to call arbitrary managed code:
+    /// It uses IL rewriting to inject a call to <c>Startup.Run(..)</c> and passes a list of assemblies.
+    /// Then, this class loads all of the specified assemblies and calls a well known etry point in those assemblies.
+    /// (See <c>TargetLibraryEntrypointXxx</c> constants in this class.)
+    /// If the specified assemblies do not contain such an entry point, they will be loaded, but nothing will be executed.
+    ///
+    /// This class sets up a basic AppDomain.AssemblyResolve handler to look for the assemblies in a framwork-specific
```

Comment:
```suggestion
    /// This class sets up a basic AppDomain.AssemblyResolve handler to look for the assemblies in a framework-specific
```

Reply:
Updated in _macrogreg/loader-with-modules-cctor-and-managed-loader_-branch.